service/build.gradle

```diff
@@ -20,249 +20,30 @@ plugins {
-    into "${jibExtraDirectory}/${cloudProfilerLocation}"
-}
-
-// OpenAPI/Swagger Server Generation
-swaggerSources {
-    server {
-        inputFile = file("${gradle.openapiSourceFile}")
-        code {
-            language = "spring"
-            library = "spring-boot"
-            outputDir = file(openapiOutputDir)
-            components = ["models", "apis"]
-            rawOptions = [
-                "--api-package", "${artifactGroup}.generated.controller",
-                "--model-package", "${artifactGroup}.generated.model",
-                "--model-name-prefix", "Api",
-                "-D", "interfaceOnly=true," +
-                      "useTags=true," +
-                      "dateLibrary=java8"
-            ]
-        }
-    }
-}
-
-// Testing config
-
-// The path to the default Google service account for the Workspace Manager to run as.
-// Created by render_config.sh
-def googleCredentialsFile = "${projectDir}/rendered/service-account.json"
-bootRun {
-    environment = [
-            'GOOGLE_APPLICATION_CREDENTIALS': "${googleCredentialsFile}"
-    ]
-}
-tasks.withType(Test) {
-    environment = [
-            'GOOGLE_APPLICATION_CREDENTIALS': "${googleCredentialsFile}"
-    ]
-}
-
-jacocoTestReport {
-    reports {
-        xml.enabled = true
-        xml.destination = file("$buildDir/reports/jacoco/test/jacoco.xml")
-    }
-    dependsOn test
-}
-
- // default test to unit tests so the build target succeeds
-test {
-    useJUnitPlatform {
-        includeTags "unit"
-    }
-}
-
-task unitTest(type: Test) {
-    useJUnitPlatform {
-        includeTags "unit"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-    finalizedBy jacocoTestReport
-}
-
-task connectedTest(type: Test) {
-    environment.put('TEST_ENV', System.getenv("TEST_ENV"))
-    useJUnitPlatform {
-        includeTags "connected"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-}
-
-
-task integrationTest(type: Test) {
-    environment.put('TEST_ENV', System.getenv("TEST_ENV"))
-    useJUnitPlatform {
-        includeTags "integration"
-    }
-    testLogging {
-        events = ["passed", "failed", "skipped", "started"]
-    }
-    outputs.upToDateWhen { false }
-}
-
-spotless {
-    java {
-        googleJavaFormat()
-        target 'src/*/java/**/*.java'
-    }
-}
-
-// Generate/clean the version properties file
-// This was being done by a plugin: https://github.com/palantir/gradle-git-version
-// but it does not work with Gradle 7.0. It was also very complicated for a
-// what we need, so easily replaced with a bash script
-def generatedVersionFile = "${resourceDir}/generated/version.properties"
-
-task generateVersionProperties(type: Exec) {
-    commandLine "${projectDir}/writeVersionProperties.sh", "${version}", "${generatedVersionFile}"
+project.ext {
```

Comment:
Maybe you can explain when to prefer `ext` and when to prefer `def`.
You changed to `ext` here, but then introduced a new `def` later.

Reply:
I belive `def` is local to the current translation unit (file), and `ext` puts an `extra property` on the project itself, so things are visible across cuts.